list/C#/List.cs

```diff
@@ -0,0 +1,195 @@
+/*
```

Comment:
I don't see a way to test this. I missed that out with some previous implementations, but we should fix that going forward. 

Implement a main method in this file that calls your implementation to ensure it is what we expect. Please consider different edge cases.